sourcesheets/views.py

```diff
@@ -1020,8 +1022,28 @@ def sheets_by_ref_api(request, ref):
     """
     API to get public sheets by ref.
     """
-    return jsonResponse(get_sheets_for_ref(ref))
+    include_collections = bool(int(request.GET.get("include_collections", 0)))
+    include_first_comment = bool(int(request.GET.get("include_first_comment", 0)))
+    return jsonResponse(get_sheets_for_ref(ref, include_collections=include_collections,
+                                           include_first_comment=include_first_comment))
+def sheets_with_ref(request, tref):
+    """
+    Accepts tref as a string which is expected to be in the format of a ref or refs separated by commas, indicating a range.
+    """
+    search_params = get_search_params(request.GET)
 
+    props={
+        "initialSearchType": "sheet",
+        "initialTextSearchField": search_params["textField"],
+        "initialSheetSearchFilters": search_params["sheetFilters"],
+        "initialSheetSearchFilterAggTypes": search_params["sheetFilterAggTypes"],
+        "initialSheetSearchSortType": search_params["sheetSort"]
+    }
+    he_tref = Ref(tref).he_normal()
+    normal_ref = tref if request.interfaceLang == "english" else he_tref
+    title = _(f"Sheets with ")+normal_ref+_(" on Sefaria")
+    props["sheetsWithRef"] = {"en": tref, "he": he_tref}
```

Comment:
Why isn't title enough? I think we should try to avoid sending this prop if possible.

Reply:
If I just send the title, how would we generate the URL when making a history state?
https://github.com/Sefaria/Sefaria-Project/blob/feature/sc-28105/re-factor-sheetsearchfilters-to-work-for/static/js/ReaderApp.jsx#L459
To generate the URL, we need access to the English ref.  If it's not passed as a prop, wouldn't we need to access window.location to know what ref we're currently on?  Also, if the hebrew ref isn't available, how would we generate the title for the history state?  We could rely on document.title.

Reply:
Ok, these are good points. Let's think about it like this: Currently this page can only load on fresh load as far as I understand but theoretically it could be accessed in the app. How would the app know this information if you navigate in-app? You would need state on ReaderApp presumably.